### PR TITLE
Migrate framework dependencies for agentd ITs

### PR DESCRIPTION
**Pull request by mjcr99**
|Related issue|
|---|
|Epic: https://github.com/wazuh/wazuh/issues/17427|
|Subtask: https://github.com/wazuh/wazuh/issues/17696

## Description
This PR includes all the necessary changes to give support to the migrated tests in the wazuh/wazuh repository. These tests will use this new structure. This PR removes all the unnecessary utilities, functions, classes, etc. that were not being used by the old ITs and adds some modifications.

### CODE REVIEW

File: src/wazuh_testing/modules/agentd/configuration.py [hunk 1]
```diff
@@ -4,3 +4,4 @@
 
 # Internal configuration options
 AGENTD_WINDOWS_DEBUG = 'windows.debug'
+AGENTD_DEBUG= 'agent.debug'
```
Comment (TomasTurina) [inline src/wazuh_testing/modules/agentd/configuration.py:7]: Add new line at the end of the file.

**Review by TomasTurina — CHANGES_REQUESTED** [review]

File: src/wazuh_testing/modules/agentd/configuration.py [hunk 2]
```diff
@@ -4,3 +4,5 @@
 
 # Internal configuration options
 AGENTD_WINDOWS_DEBUG = 'windows.debug'
+AGENTD_DEBUG = 'agent.debug'
+AGENTD_TIMEOUT = 'agent.recv_timeout'
```
Comment (TomasTurina) [inline src/wazuh_testing/modules/agentd/configuration.py:8]: Add new line at the end of the file.

**Comment by TomasTurina** [comment]
Please, rebase your PR to add all latest changes.

File: src/wazuh_testing/modules/agentd/configuration.py (hunk 1 above)
Comment (mjcr99) [reply src/wazuh_testing/modules/agentd/configuration.py]: Fixed with https://github.com/wazuh/qa-integration-framework/pull/31/commits/c2fe2409c925a501c02dee095e309f1b85631337

File: src/wazuh_testing/modules/agentd/configuration.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mjcr99) [reply src/wazuh_testing/modules/agentd/configuration.py]: Fixed with https://github.com/wazuh/qa-integration-framework/pull/31/commits/c2fe2409c925a501c02dee095e309f1b85631337

File: src/wazuh_testing/tools/mitm.py [hunk 3]
```diff
@@ -224,7 +231,6 @@ def handle(self):
             self.server.mitm.handler_func(data)
             self.server.mitm.put_queue(data)
```
[the remark this answers is not in the document]
Comment (mjcr99) [reply src/wazuh_testing/tools/mitm.py]: Fixed with https://github.com/wazuh/qa-integration-framework/pull/31/commits/c2fe2409c925a501c02dee095e309f1b85631337

**Review by TomasTurina — APPROVED** [review]

**Review by TomasTurina — APPROVED** [review]

**Pull request merged**